### ISSUE TITLE
weights sometimes are being written as zeros

### ISSUE THREAD

**Issue by Joshuaalbert** [revoltek/losoto#69]
I have an extremely perplexing case. I have an array that when I write to weights of a soltab and then read again they are zeros, but when I take a random numpy array of the same shape and dtype it works. Here is an example of the code.

Any ideas what could be causing one array to work and not another. I've also tried `array.copy()` before setting values just to make sure it's not a memory location thing.

## MVCE
``` python 
# array comes from a solve is float16 and same shape as below weights.shape
# array is non zero

assert np.all(np.isfinite(array)) == True

soltab = H.getSoltab('tec000')
soltab.clearSelection()
soltab.setValues(array, weight=True)

weights = soltab.getValues(weight=True)
#weights are zero

soltab.setValues(np.random.uniform(size=weights.shape), weight=True)

weights = soltab.getValues(weight=True)
#weights are non-zero
```

**Comment by revoltek** [comment]
Weights should be between 0 (==flagged) and 1. They are also np.float16, not np.float64 like solutions, this might be the root of the problem.

**Comment by Joshuaalbert** [comment]
Solved and still perplexed. arrays has tec variances which are on order of 1e-6 and np.float16 properly represents this but for some reason the smallest number that losoto is storing is 4 significant figures, which truncates everything to zeros.

**Comment by Joshuaalbert** [comment]
I'm using the weight column to store uncertainties for the values (abusing the original use but it saves space rather than making a new soltab). Also I'm surprised the precision isn't the same as np.float16.

**Comment by Joshuaalbert** [comment]
@revoltek do you think it would be simple to include a flag that chosses the dtype of weights (with default set to float16 of course so it's backward compatible)?

**Comment by Joshuaalbert** [comment]
The only point the dtype of weights is chosen is https://github.com/revoltek/losoto/blob/master/losoto/h5parm.py#L510

If we include a `weight_dtype` kwarg in makeSoltab that defaults to f16 it should be a simple fix, then I could choose float32 or 64 for my work. I'll make a PR.

**Comment by Joshuaalbert** [comment]
Solved by b747a76b8fa8b557f662e77e0f6bbe68d2dd0dde
